extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -156,17 +183,74 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
   }
 
   /** @private */
-  onPageNavigation_() {
-    if (this.isOnActivePage_) {
-      this.animate_();
+  setAnimateTo_() {
+    const x = parseFloat(this.element_.getAttribute('x') || 0);
+    const y = parseFloat(this.element_.getAttribute('y') || 0);
+    const zoom = parseFloat(this.element_.getAttribute('zoom') || 1);
+    const lockBounds = this.element_.hasAttribute('lock-bounds');
+
+    if (!lockBounds) {
+      this.animateTo_ = {x, y, zoom};
+    } else {
+      // Zoom must be set to calculate maxBounds.
+      this.animateTo_.zoom = zoom < 1 ? 1 : zoom;
+
+      this.setMaxBounds_();
```

Comment:
maxBounds doesn't need to be a private variable if used like this, as you could just return the JSON and set it to a const here. If it were to be cached, then using the private variable might make sense, but not if the usecase can be replaced with a `const maxBounds = this.calculateMaxBounds_()`